portal-bridge/src/beacon_bridge.rs

```diff
@@ -0,0 +1,237 @@
+use crate::consensus_api::ConsensusApi;
+use crate::constants::BEACON_GENESIS_TIME;
+use crate::mode::BridgeMode;
+use crate::utils::{duration_until_next_update, expected_current_slot};
+use ethportal_api::types::consensus::fork::ForkName;
+use ethportal_api::types::consensus::light_client::bootstrap::LightClientBootstrapCapella;
+use ethportal_api::types::consensus::light_client::finality_update::LightClientFinalityUpdateCapella;
+use ethportal_api::types::consensus::light_client::optimistic_update::LightClientOptimisticUpdateCapella;
+use ethportal_api::types::consensus::light_client::update::{
+    LightClientUpdate, LightClientUpdateCapella,
+};
+use ethportal_api::types::content_key::beacon::ZeroKey;
+use ethportal_api::types::content_value::beacon::{
+    ForkVersionedLightClientUpdate, LightClientUpdatesByRange,
+};
+use ethportal_api::{BeaconContentKey, BeaconContentValue, LightClientUpdatesByRangeKey};
+use ethportal_api::{BeaconNetworkApiClient, ContentValue};
+use jsonrpsee::http_client::HttpClient;
+use serde_json::Value;
+use ssz::Encode;
+use ssz_types::VariableList;
+use std::cmp::Ordering;
+use std::sync::atomic::AtomicUsize;
+use std::sync::Arc;
+use std::thread::sleep;
+use tracing::{info, warn};
+
+pub struct BeaconBridge {
+    pub api: ConsensusApi,
+    mode: BridgeMode,
+    portal_clients: Vec<HttpClient>,
+}
+
+impl BeaconBridge {
+    pub fn new(api: ConsensusApi, mode: BridgeMode, portal_clients: Vec<HttpClient>) -> Self {
```

Comment:
Something to think about, but not urgent to address is how you'd like to do metrics here? Can they both use the same shared types? And how `BeaconBridge` metrics will be reported alongside the regular `Bridge` metrics.